openvm/src/customize_exe.rs

```diff
@@ -453,197 +488,3 @@ pub fn openvm_bus_interaction_to_powdr<F: PrimeField32, P: FieldElement>(
 
     Ok(SymbolicBusInteraction { id, mult, args })
 }
-
-// Note: This function can lead to OOM since it generates the apc for many blocks.
-fn create_apcs_with_cell_pgo<P: IntoOpenVm>(
-    mut blocks: Vec<BasicBlock<OpenVmField<P>>>,
-    pgo_program_idx_count: HashMap<u32, u32>,
-    max_total_columns: Option<usize>,
-    airs: &OriginalAirs<P>,
-    config: &PowdrConfig,
-    original_config: &OriginalVmConfig,
-    bus_map: &BusMap,
-) -> Vec<BlockWithApc<P>> {
-    // drop any block whose start index cannot be found in pc_idx_count,
-    // because a basic block might not be executed at all.
-    // Also only keep basic blocks with more than one original instruction.
-    blocks.retain(|b| {
```

Comment:
We don't need to do this here. We can do it during the single iteration on the blocks lower down.